explorer/app/components/TableCreator/tableCreator.tsx

```diff
@@ -32,21 +32,81 @@ export const TableCreator = <T extends object>({
   pagination,
   sort,
 }: TableCreatorProps<T>): JSX.Element => {
-  const reactColumns: Column<T>[] = columns.map((columnConfig) => ({
-    Cell: createCell(columnConfig),
-    Header: columnConfig.header,
-    disableSortBy: !columnConfig.sort,
-    id: columnConfig.sort?.sortKey,
+  const defaultColumns = useMemo(
```

Comment:
Is there a way to encapsulate the edit column functionality defined in this area, maybe a hook? I don't think we'd be using the hook outside of this `tableCreator` but it might be helpful for readability and testability to have the edit column functionality "isolated".